syntax = "proto3";

package finazon;

// Crypto

message GetMarketsCryptoRequest {
  int32  page = 1;
  int32  page_size = 2;
}

message MarketsCryptoResponseItem {
  string code = 1;
  string name = 2;
}

message GetMarketsCryptoResponse {
  repeated MarketsCryptoResponseItem result = 1;
}

// Stocks

message GetMarketsStocksRequest {
  string country = 1;
  string name = 2;
  string code = 3;
  int32  page = 4;
  int32  page_size = 5;
}

message MarketsStocksResponseItem {
  string country = 1;
  string mic = 2;
  string name = 3;
}

message GetMarketsStocksResponse {
  repeated MarketsStocksResponseItem result = 1;
}

/**
 * ExchangeService provide access to Finazon exchange data
 */
service ExchangeService {
  // Returns a list of supported crypto markets
  rpc GetMarketsCrypto(GetMarketsCryptoRequest) returns (GetMarketsCryptoResponse) {};

  // Returns a list of supported stock markets
  rpc GetMarketsStocks(GetMarketsStocksRequest) returns (GetMarketsStocksResponse) {};
}
